Code/GraphMol/SubstructLibrary/SubstructLibrary.h

```diff
@@ -753,6 +806,17 @@ class RDKIT_SUBSTRUCTLIBRARY_EXPORT SubstructLibrary {
   const std::vector<unsigned int> &getSearchOrder() const {
     return searchOrder;
   }
+  const std::string &getKey(unsigned int idx) const {
+    // expects implementation to throw IndexError if out of range
+    PRECONDITION(keyholder.get() != nullptr, "SubstructLibrary Has no keys");
```

Comment:
`PRECONDITION`s are for cases which really shouldn't happen or where the programmer has made a probably non-recoverable mistake.
Given that it's possible (and acceptable) to create a library without a keyholder, this should be a `ValueError`

Reply:
This is now a value error through the use of getKeys() and returning the key holder.